internal/integration/csot_prose_test.go

```diff
@@ -177,6 +177,63 @@ func TestCSOTProse(t *testing.T) {
 				"expected ping to fail within 150ms")
 		})
 	})
+
+	mt.RunOpts("11. multi-batch bulkWrites", mtest.NewOptions().MinServerVersion("8.0").
+		AtlasDataLake(false).Topologies(mtest.Single), func(mt *mtest.T) {
+		coll := mt.CreateCollection(mtest.Collection{DB: "db", Name: "coll"}, false)
+		err := coll.Drop(context.Background())
+		require.NoError(mt, err, "Drop error: %v", err)
+
+		mt.SetFailPoint(failpoint.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: failpoint.Mode{
+				Times: 2,
+			},
+			Data: failpoint.Data{
+				FailCommands:    []string{"bulkWrite"},
+				BlockConnection: true,
+				BlockTimeMS:     1010,
+			},
+		})
+
+		var hello struct {
+			MaxBsonObjectSize   int
+			MaxMessageSizeBytes int
+		}
+		err = mt.DB.RunCommand(context.Background(), bson.D{{"hello", 1}}).Decode(&hello)
+		require.NoError(mt, err, "Hello error: %v", err)
+
+		var writes []mongo.ClientBulkWrite
+		n := hello.MaxMessageSizeBytes/hello.MaxBsonObjectSize + 1
+		for i := 0; i < n; i++ {
+			writes = append(writes, mongo.ClientBulkWrite{
+				Database:   "db",
+				Collection: "coll",
+				Model: &mongo.ClientInsertOneModel{
+					Document: bson.D{{"a", strings.Repeat("b", hello.MaxBsonObjectSize-500)}},
+				},
+			})
+		}
+
+		var cnt int
+		cm := &event.CommandMonitor{
+			Started: func(_ context.Context, evt *event.CommandStartedEvent) {
+				if evt.CommandName == "bulkWrite" {
+					cnt++
+				}
+			},
+		}
+		cliOptions := options.Client().
+			SetTimeout(2 * time.Second).
+			SetMonitor(cm).
+			ApplyURI(mtest.ClusterURI())
+		integtest.AddTestServerAPIVersion(cliOptions)
+		cli, err := mongo.Connect(cliOptions)
+		require.NoError(mt, err, "Connect error: %v", err)
+		_, err = cli.BulkWrite(context.Background(), writes)
```

Comment:
Use `mt.ResetClient` instead.

```suggestion
		cliOptions := options.Client().
			SetTimeout(2 * time.Second).
			SetMonitor(cm)
		mt.ResetClient(cliOptions)
		_, err = mt.Client.BulkWrite(context.Background(), writes)
```